src/CoreBundle/DataAccess/CmsPortalDomainsDataAccessRequestLevelCacheDecorator.php

```diff
@@ -47,4 +47,28 @@ public function getPrimaryDomain($portalId, $languageId)
 
         return $domain;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPortalPrefixListForDomain(string $domainName): array
+    {
+        return $this->subject->getPortalPrefixListForDomain($domainName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getActivePortalCandidate(array $idRestrictionList, string $identifierRestriction, bool $allowInactivePortals): ?array
+    {
+        return $this->subject->getActivePortalCandidate($idRestrictionList, $identifierRestriction, $allowInactivePortals);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDomainDataByName(string $domainName): array
+    {
+        return $this->subject->getDomainDataByName($domainName);
+    }
```

Comment:
There is not much request level caching going on in here. Is this intentionally?

Reply:
This comment isn't shown in the discussion ..?

This is indeed by intention - these methods are quite specific for initialization and are only called once per request. Adding a request level cache wouldn't improve performance but cost additional memory, so I didn't do this.